lib/endpoints/pivtokens.js

```diff
@@ -401,9 +403,14 @@ function updateRecoveryTokens(req, res, next) {
     opts.moray = req.app.moray;
     opts.log = req.log;
 
-    const func = opts.active_recovery_config_uuid &&
-                    opts.staged_recovery_config_uuid ? 'setActiveAndStaged' :
-                opts.staged_recovery_config_uuid ? 'setStaged' : 'setActive';
+    // If `staged` is present, `active` will be different, which means we've
+    // just staged a new recovery config and we still have the originally
+    // active one.
+    // When we activate a recovery configuration, it must be already staged and
+    // from the cn-agent, we'll then remove the `staged` member from
+    // `Zpool Recovery` sysinfo property. Therefore, if `staged` is not present,
+    // we are doing an activation.
+    const func = opts.staged_recovery_config_uuid ? 'setStaged' : 'setActive';
```

Comment:
Also verified with @jasonbking that this is the current behavior of the OS/kbmd pieces.